packages/mimir/src/rules/parameter-properties.ts

```diff
@@ -0,0 +1,193 @@
+    }
+
+    return false;
+}
+
+function getFixerForDisallowedParameterProp(
+    construct: ts.ConstructorDeclaration,
+    param: ts.ParameterDeclaration,
+    lineBreak: string,
+): Replacement[] {
+    const superCall = getSuperCall(construct);
+    return [
+        /* Add assignment after super() call (if it exists) */
+        Replacement.append(
+            superCall ? superCall.end : construct.body!.getStart() + 1,
+            `${lineBreak}this.${param.name.getText()} = ${param.name.getText()};`,
+        ),
+
+        /* Add properties to class, trimming off default values if necessary */
+        Replacement.append(
+            construct.getStart(),
+            (param.getText().indexOf('=') > -1
+                ? param.getText().substring(0, param.getText().indexOf('=') - 1)
+                : param.getText()) +
+                ';' +
+                lineBreak,
+        ),
+
+        /* Finally, delete modifiers from the parameter prop */
+        Replacement.delete(param.pos, param.name.getStart()),
+    ];
+}
+
+function getFixerForLonghandProp(
+    param: ts.ParameterDeclaration,
+    construct: ts.ConstructorDeclaration,
+    classNode: ts.ClassDeclaration | ts.ClassExpression,
+): Replacement[] {
+    /* Class member and assignment to be removed */
+    const member = classNode.members
+        .filter(isPropertyDeclaration)
+        .find((prop) => prop.name.getText() === param.name.getText())!;
+    const assignment = construct
+        .body!.statements.filter(isExpressionStatement)
+        .find((stmt) => isSimpleParamToPropAssignment(stmt, param))!;
+
+    return [
+        Replacement.delete(member.getStart(), member.end),
+        Replacement.delete(assignment.getStart(), assignment.end),
+
+        /* Append access modifiers to parameter declaration */
+        Replacement.append(
+            param.getStart(),
+            member.modifiers ? member.modifiers.map((mod) => mod.getText()).join(' ') + ' ' : 'public ',
+        ),
+    ];
+}
+
+function getSuperCall(construct: ts.ConstructorDeclaration): ts.ExpressionStatement | undefined {
+    const firstStmt = construct.body!.statements.length > 0 ? construct.body!.statements[0] : undefined;
+
+    return firstStmt &&
+        isExpressionStatement(firstStmt) &&
+        isCallExpression(firstStmt.expression) &&
+        firstStmt.expression.expression.kind === ts.SyntaxKind.SuperKeyword
+        ? firstStmt
+        : undefined;
+}
+
+function isSimpleParamToPropAssignment(stmt: ts.ExpressionStatement, param: ts.ParameterDeclaration): boolean {
+    return (
+        isBinaryExpression(stmt.expression) &&
+        isPropertyAccessExpression(stmt.expression.left) &&
+        stmt.expression.left.name.text === param.name.getText() &&
+        isIdentifier(stmt.expression.right) &&
+        stmt.expression.right.text === param.name.getText() &&
+        !hasSideEffects(stmt.expression.left) &&
+        !hasSideEffects(stmt.expression.right)
```

Comment:
if you know what `left` and `right` are, you don't need to check for side effects anymore